lib/elixir/lib/calendar/date.ex

```diff
@@ -810,6 +810,50 @@ defmodule Date do
     shift(date, Duration.new!(duration))
   end
 
+  @doc """
+  Returns the current date in UTC subtracting the given `duration`.
+
+  ## Examples
+
+      iex> date = Date.utc_ago(day: 1)
+      iex> date == Date.add(Date.utc_today(), -1)
+
+      iex> date = Date.utc_ago(month: 1)
+      iex> date == Date.shift(Date.utc_today(), month: -1)
+
+  """
+  @doc since: "1.17.0"
+  @spec utc_ago(Duration.t() | [Duration.unit_pair()]) :: t
```

Comment:
I dont think it was but it seems like a reasonable addition! Might become easier to reason about once we added more functions supporting Durations

Reply:
I think this type is good and I would add it to Duration itself. We should probably make it so `Duration.new!/1` also accepts an existing duration and just return it. This way we can simplify functions such as `shift` to just call `Duration.new!`? This can be done separately from this PR. :)

Reply:
here it is: https://github.com/elixir-lang/elixir/pull/13476 thanks for the input!